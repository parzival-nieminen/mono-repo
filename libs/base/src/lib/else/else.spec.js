import {render} from '@testing-library/react';
import Else from './else';
import renderer from "react-test-renderer";

describe('Else', () => {
  it('should render successfully', () => {
    const {baseElement} = render(<Else/>);
    expect(baseElement).toBeTruthy();
  });

  it('should render successfully', () => {
    const tree = renderer.create(<Else/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('show text: Hello World', () => {
    const tree = renderer.create(<Else>Hello World</Else>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('show text: Hello World my friend', () => {
    const {baseElement} = render(<Else>
      Hello
      <div> World, </div>
      <div><p>my friend</p></div>
    </Else>);
    expect(baseElement.textContent.trim()).toBe("Hello World, my friend");
  });

});
