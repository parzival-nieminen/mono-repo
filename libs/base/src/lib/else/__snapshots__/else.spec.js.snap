// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Else should render successfully 1`] = `<div />`;

exports[`Else show text: Hello World 1`] = `
<div>
  Hello World
</div>
`;
